lib/cartopy/io/shapereader.py

```diff
@@ -394,13 +400,38 @@ def default_downloader():
         return NEShpDownloader(target_path_template=ne_path_template,
                                pre_downloaded_path_template=pre_path_template)
 
+    @staticmethod
+    def versioned_downloader():
+        """
+        Return a NEShpDownloader instance, which extends the default downloader
+        with support for specific file versions.
+
+        """
+        _NE_URL_TEMPLATE = ('https://naturalearth.s3.amazonaws.com/'
+                            '{version}/{resolution}_{category}/'
+                            'ne_{resolution}_{name}.zip')
+
+        default_spec = ('shapefiles', 'natural_earth', '{category}', '{version}',
+                        'ne_{resolution}_{name}.shp')
```

Comment:
Would you be able to inject this into the current NE template? It looks like the only difference is `{version}/`, and I'm wondering if you could put that in and always have a version with something like `_version_string = "" if version is None else f"{version}/".

Reply:
Yes, I couldn't figure out how to inject version without making the second default downloader, but I think this solution works! Only problem is it obscures the template slightly but worth it to avoid duplicating the other code. For the other question, yes, if latest is 5.1.2 and then you specifically ask for 5.1.2 it will download it twice, since version-specified data will go in its own subfolder (within the relevant physical or cultural folder). Unfortunately, the .VERSION file is currently filtered out so it is hard to tell what version you have downloaded in the base folder.